### PR TITLE
Add more conditional to the unittest that have caused the parser to fail

### PR DESCRIPTION
**Pull request by matthewfcarlson**

### CODE REVIEW

File: edk2toollib/uefi/edk2/parsers/base_parser.py [hunk 1]
```diff
@@ -438,60 +482,90 @@ def EvaluateConditional(self, text):
...
+            # pop the stack and print the top operator.
+            # Then test the incoming operator against the new top of stack.
+            elif len(stack) != 0 and cls._GetOperatorPrecedence(token) < cls._GetOperatorPrecedence(stack[-1]):
+                while len(stack) > 0 and cls._GetOperatorPrecedence(token) <= cls._GetOperatorPrecedence(stack[-1]):
                     expression.append(stack.pop())
                 stack.append(token)
             else:
-                expression.append(token)
+                logging.error("We don't know what to do with " + token)
+            logging.debug(f"{token.ljust(10)} | {' '.join(stack).ljust(50)} | {' '.join(expression).ljust(50)}")
```
Comment (spbrogan) [inline edk2toollib/uefi/edk2/parsers/base_parser.py:544]: what is "50".  
Can you add a comment for the magic number usage

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/base_parser.py:544]: That's just padding it to a new tabular format. It's pretty arbitrary.

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/base_parser.py:544]: I might just take this out

**Comment by matthewfcarlson** [comment]
This is ready to go if you're happy with it

**Pull request merged**